File: server/game/maps.go
package game

var Maps = map[string]GameMap{
	"outpost": {
		ID:              "outpost",
		Name:            "Outpost (Classic)",
		Width:           2000,
		Height:          1200,
		BackgroundColor: "#1e272c",
		SkyColor:        "#2c3e50",
		Platforms: []Platform{
			// Outer boundaries / Ground
			{X: 0, Y: 1140, W: 2000, H: 60, Type: "rock", Color: "#34495e"},
			{X: 0, Y: 0, W: 40, H: 1200, Type: "rock", Color: "#2c3e50"},
			{X: 1960, Y: 0, W: 40, H: 1200, Type: "rock", Color: "#2c3e50"},
			{X: 0, Y: 0, W: 2000, H: 40, Type: "rock", Color: "#2c3e50"},

			// Main Center Bunker / Outpost Tower
			{X: 750, Y: 700, W: 500, H: 30, Type: "metal", Color: "#7f8c8d"},
			{X: 800, Y: 480, W: 400, H: 25, Type: "metal", Color: "#95a5a6"},
			{X: 850, Y: 280, W: 300, H: 25, Type: "metal", Color: "#bdc3c7"},

			// Center Support Pillars & Stairways
			{X: 880, Y: 730, W: 30, H: 410, Type: "metal", Color: "#7f8c8d"},
			{X: 1090, Y: 730, W: 30, H: 410, Type: "metal", Color: "#7f8c8d"},

			// Left Base Structure & Bridges
			{X: 200, Y: 920, W: 350, H: 25, Type: "metal", Color: "#95a5a6"},
			{X: 120, Y: 680, W: 300, H: 25, Type: "metal", Color: "#7f8c8d"},
			{X: 250, Y: 440, W: 300, H: 25, Type: "one-way", Color: "#bdc3c7"},

			// Right Base Structure & Bridges
			{X: 1450, Y: 920, W: 350, H: 25, Type: "metal", Color: "#95a5a6"},
			{X: 1580, Y: 680, W: 300, H: 25, Type: "metal", Color: "#7f8c8d"},
			{X: 1450, Y: 440, W: 300, H: 25, Type: "one-way", Color: "#bdc3c7"},

			// High Sniper Platforms
			{X: 450, Y: 220, W: 200, H: 20, Type: "one-way", Color: "#e74c3c"},
			{X: 1350, Y: 220, W: 200, H: 20, Type: "one-way", Color: "#e74c3c"},

			// Underground Tunnel Bunkers
			{X: 500, Y: 1050, W: 300, H: 20, Type: "one-way", Color: "#34495e"},
			{X: 1200, Y: 1050, W: 300, H: 20, Type: "one-way", Color: "#34495e"},
		},
		Bushes: []Bush{
			{X: 860, Y: 235, W: 120, H: 45},
			{X: 280, Y: 395, W: 140, H: 45},
			{X: 1500, Y: 395, W: 140, H: 45},
			{X: 220, Y: 875, W: 160, H: 45},
			{X: 1520, Y: 875, W: 160, H: 45},
			{X: 920, Y: 1095, W: 160, H: 45},
		},
		Spawns: []Vector2D{
			{X: 250, Y: 620},
			{X: 1750, Y: 620},
			{X: 1000, Y: 220},
			{X: 1000, Y: 640},
			{X: 300, Y: 1080},
			{X: 1700, Y: 1080},
			{X: 500, Y: 160},
			{X: 1500, Y: 160},
		},
		WeaponSpawns: []WeaponSpawn{
			{X: 1000, Y: 250, WeaponType: "sniper"},
			{X: 1000, Y: 660, WeaponType: "ar"},
			{X: 280, Y: 640, WeaponType: "smg"},
			{X: 1720, Y: 640, WeaponType: "ar"},
			{X: 350, Y: 880, WeaponType: "smg"},
			{X: 1600, Y: 880, WeaponType: "sniper"},
			{X: 500, Y: 180, WeaponType: "ar"},
			{X: 1450, Y: 180, WeaponType: "smg"},
			{X: 600, Y: 1010, WeaponType: "sniper"},
			{X: 1350, Y: 1010, WeaponType: "ar"},
		},
		HealthSpawns: []Vector2D{
			{X: 1000, Y: 440},
			{X: 150, Y: 640},
			{X: 1800, Y: 640},
			{X: 1000, Y: 1080},
		},
		BoosterSpawns: []Vector2D{
			{X: 880, Y: 440},
			{X: 1120, Y: 440},
			{X: 270, Y: 640},
			{X: 1680, Y: 640},
		},
	},
	"catacombs": {
		ID:              "catacombs",
		Name:            "Catacombs (Cavern)",
		Width:           2200,
		Height:          1400,
		BackgroundColor: "#1b120c",
		SkyColor:        "#2c1810",
		Platforms: []Platform{
			// Outer rock borders
			{X: 0, Y: 1340, W: 2200, H: 60, Type: "rock", Color: "#4a3525"},
			{X: 0, Y: 0, W: 50, H: 1400, Type: "rock", Color: "#4a3525"},
			{X: 2150, Y: 0, W: 50, H: 1400, Type: "rock", Color: "#4a3525"},
			{X: 0, Y: 0, W: 2200, H: 50, Type: "rock", Color: "#4a3525"},

			// Central cavern rock arch & shafts
			{X: 800, Y: 800, W: 600, H: 40, Type: "rock", Color: "#5d4037"},
			{X: 650, Y: 550, W: 900, H: 35, Type: "rock", Color: "#5d4037"},
			{X: 900, Y: 320, W: 400, H: 30, Type: "rock", Color: "#6d4c41"},

			// Left cavern chambers
			{X: 150, Y: 1100, W: 450, H: 30, Type: "rock", Color: "#5d4037"},
			{X: 200, Y: 820, W: 350, H: 30, Type: "rock", Color: "#5d4037"},
			{X: 100, Y: 500, W: 400, H: 30, Type: "rock", Color: "#5d4037"},
			{X: 250, Y: 250, W: 300, H: 25, Type: "one-way", Color: "#8d6e63"},

			// Right cavern chambers
			{X: 1600, Y: 1100, W: 450, H: 30, Type: "rock", Color: "#5d4037"},
			{X: 1650, Y: 820, W: 350, H: 30, Type: "rock", Color: "#5d4037"},
			{X: 1700, Y: 500, W: 400, H: 30, Type: "rock", Color: "#5d4037"},
			{X: 1650, Y: 250, W: 300, H: 25, Type: "one-way", Color: "#8d6e63"},

			// Vertical shaft dividers
			{X: 600, Y: 830, W: 35, H: 300, Type: "rock", Color: "#4a4a25"},
			{X: 1565, Y: 830, W: 35, H: 300, Type: "rock", Color: "#4a4a25"},
		},
		Bushes: []Bush{
			{X: 1020, Y: 280, W: 160, H: 40},
			{X: 1020, Y: 510, W: 160, H: 40},
			{X: 320, Y: 780, W: 140, H: 40},
			{X: 1720, Y: 780, W: 140, H: 40},
			{X: 980, Y: 1300, W: 240, H: 40},
		},
		Spawns: []Vector2D{
			{X: 300, Y: 440},
			{X: 1900, Y: 440},
			{X: 1100, Y: 260},
			{X: 1100, Y: 740},
			{X: 350, Y: 1040},
			{X: 1850, Y: 1040},
		},
		WeaponSpawns: []WeaponSpawn{
			{X: 1100, Y: 280, WeaponType: "sniper"},
			{X: 1100, Y: 510, WeaponType: "ar"},
			{X: 300, Y: 460, WeaponType: "smg"},
			{X: 1900, Y: 460, WeaponType: "sniper"},
			{X: 350, Y: 1060, WeaponType: "ar"},
			{X: 1850, Y: 1060, WeaponType: "smg"},
			{X: 1100, Y: 760, WeaponType: "ar"},
		},
		HealthSpawns: []Vector2D{
			{X: 1100, Y: 480},
			{X: 250, Y: 780},
			{X: 1950, Y: 780},
		},
	},
	"hightower": {
		ID:              "hightower",
		Name:            "High Tower (Sky Battle)",
		Width:           2400,
		Height:          1500,
		BackgroundColor: "#0c1b2b",
		SkyColor:        "#1a365d",
		Platforms: []Platform{
			// Bottom Void Danger & Base Platform
			{X: 0, Y: 1440, W: 2400, H: 60, Type: "rock", Color: "#1a202c"},
			{X: 0, Y: 0, W: 40, H: 1500, Type: "rock", Color: "#1a202c"},
			{X: 2360, Y: 0, W: 40, H: 1500, Type: "rock", Color: "#1a202c"},

			// Center High Tower Structure
			{X: 1050, Y: 200, W: 300, H: 1200, Type: "metal", Color: "#2d3748"},
			{X: 950, Y: 200, W: 500, H: 30, Type: "metal", Color: "#4a5568"},
			{X: 900, Y: 450, W: 600, H: 30, Type: "metal", Color: "#4a5568"},
			{X: 850, Y: 700, W: 700, H: 30, Type: "metal", Color: "#4a5568"},
			{X: 800, Y: 950, W: 800, H: 30, Type: "metal", Color: "#4a5568"},
			{X: 750, Y: 1200, W: 900, H: 35, Type: "metal", Color: "#4a5568"},

			// Side Floating Sky Platforms (Left)
			{X: 250, Y: 350, W: 350, H: 25, Type: "one-way", Color: "#3182ce"},
			{X: 150, Y: 650, W: 400, H: 25, Type: "one-way", Color: "#3182ce"},
			{X: 300, Y: 950, W: 350, H: 25, Type: "one-way", Color: "#3182ce"},

			// Side Floating Sky Platforms (Right)
			{X: 1800, Y: 350, W: 350, H: 25, Type: "one-way", Color: "#3182ce"},
			{X: 1850, Y: 650, W: 400, H: 25, Type: "one-way", Color: "#3182ce"},
			{X: 1750, Y: 950, W: 350, H: 25, Type: "one-way", Color: "#3182ce"},
		},
		Bushes: []Bush{
			{X: 1120, Y: 160, W: 160, H: 40},
			{X: 350, Y: 310, W: 140, H: 40},
			{X: 1910, Y: 310, W: 140, H: 40},
			{X: 1120, Y: 660, W: 160, H: 40},
			{X: 1120, Y: 1160, W: 160, H: 40},
		},
		Spawns: []Vector2D{
			{X: 1200, Y: 140},
			{X: 400, Y: 300},
			{X: 2000, Y: 300},
			{X: 1200, Y: 640},
			{X: 300, Y: 600},
			{X: 2100, Y: 600},
			{X: 1200, Y: 1140},
		},
		WeaponSpawns: []WeaponSpawn{
			{X: 1200, Y: 160, WeaponType: "sniper"},
			{X: 1200, Y: 410, WeaponType: "ar"},
			{X: 1200, Y: 660, WeaponType: "smg"},
			{X: 400, Y: 320, WeaponType: "ar"},
			{X: 2000, Y: 320, WeaponType: "smg"},
			{X: 300, Y: 620, WeaponType: "ar"},
			{X: 2100, Y: 620, WeaponType: "sniper"},
			{X: 1200, Y: 1160, WeaponType: "smg"},
		},
		HealthSpawns: []Vector2D{
			{X: 1200, Y: 900},
			{X: 350, Y: 910},
			{X: 1950, Y: 910},
		},
	},
}

File: src/data/maps.ts
import { GameMap, MapId } from '../types';

export const MAPS: Record<MapId, GameMap> = {
  outpost: {
    id: 'outpost',
    name: 'Outpost (Classic)',
    width: 2000,
    height: 1200,
    backgroundColor: '#1e272c',
    skyColor: '#2c3e50',
    platforms: [
      // Outer boundaries / Ground
      { x: 0, y: 1140, w: 2000, h: 60, type: 'rock', color: '#34495e' },
      { x: 0, y: 0, w: 40, h: 1200, type: 'rock', color: '#2c3e50' },
      { x: 1960, y: 0, w: 40, h: 1200, type: 'rock', color: '#2c3e50' },
      { x: 0, y: 0, w: 2000, h: 40, type: 'rock', color: '#2c3e50' },

      // Main Center Bunker / Outpost Tower
      { x: 750, y: 700, w: 500, h: 30, type: 'metal', color: '#7f8c8d' },
      { x: 800, y: 480, w: 400, h: 25, type: 'metal', color: '#95a5a6' },
      { x: 850, y: 280, w: 300, h: 25, type: 'metal', color: '#bdc3c7' },

      // Center Support Pillars & Stairways
      { x: 880, y: 730, w: 30, h: 410, type: 'metal', color: '#7f8c8d' },
      { x: 1090, y: 730, w: 30, h: 410, type: 'metal', color: '#7f8c8d' },

      // Left Base Structure & Bridges
      { x: 200, y: 920, w: 350, h: 25, type: 'metal', color: '#95a5a6' },
      { x: 120, y: 680, w: 300, h: 25, type: 'metal', color: '#7f8c8d' },
      { x: 250, y: 440, w: 300, h: 25, type: 'one-way', color: '#bdc3c7' },

      // Right Base Structure & Bridges
      { x: 1450, y: 920, w: 350, h: 25, type: 'metal', color: '#95a5a6' },
      { x: 1580, y: 680, w: 300, h: 25, type: 'metal', color: '#7f8c8d' },
      { x: 1450, y: 440, w: 300, h: 25, type: 'one-way', color: '#bdc3c7' },

      // High Sniper Platforms
      { x: 450, y: 220, w: 200, h: 20, type: 'one-way', color: '#e74c3c' },
      { x: 1350, y: 220, w: 200, h: 20, type: 'one-way', color: '#e74c3c' },

      // Underground Tunnel Bunkers
      { x: 500, y: 1050, w: 300, h: 20, type: 'one-way', color: '#34495e' },
      { x: 1200, y: 1050, w: 300, h: 20, type: 'one-way', color: '#34495e' }
    ],
    bushes: [
      { x: 860, y: 235, w: 120, h: 45 },
      { x: 280, y: 395, w: 140, h: 45 },
      { x: 1500, y: 395, w: 140, h: 45 },
      { x: 220, y: 875, w: 160, h: 45 },
      { x: 1520, y: 875, w: 160, h: 45 },
      { x: 920, y: 1095, w: 160, h: 45 }
    ],
    spawns: [
      { x: 250, y: 620 },
      { x: 1750, y: 620 },
      { x: 1000, y: 220 },
      { x: 1000, y: 640 },
      { x: 300, y: 1080 },
      { x: 1700, y: 1080 },
      { x: 500, y: 160 },
      { x: 1500, y: 160 }
    ],
    weaponSpawns: [
      { x: 1000, y: 250, weaponType: 'sniper' },
      { x: 1000, y: 660, weaponType: 'ar' },
      { x: 280, y: 640, weaponType: 'smg' },
      { x: 1720, y: 640, weaponType: 'ar' },
      { x: 350, y: 880, weaponType: 'smg' },
      { x: 1600, y: 880, weaponType: 'sniper' },
      { x: 500, y: 180, weaponType: 'ar' },
      { x: 1450, y: 180, weaponType: 'smg' },
      { x: 600, y: 1010, weaponType: 'sniper' },
      { x: 1350, y: 1010, weaponType: 'ar' }
    ],
    healthSpawns: [
      { x: 1000, y: 440 },
      { x: 150, y: 640 },
      { x: 1800, y: 640 },
      { x: 1000, y: 1080 }
    ]
  },
  catacombs: {
    id: 'catacombs',
    name: 'Catacombs (Cavern)',
    width: 2200,
    height: 1400,
    backgroundColor: '#1b120c',
    skyColor: '#2c1810',
    platforms: [
      // Outer rock borders
      { x: 0, y: 1340, w: 2200, h: 60, type: 'rock', color: '#4a3525' },
      { x: 0, y: 0, w: 50, h: 1400, type: 'rock', color: '#4a3525' },
      { x: 2150, y: 0, w: 50, h: 1400, type: 'rock', color: '#4a3525' },
      { x: 0, y: 0, w: 2200, h: 50, type: 'rock', color: '#4a3525' },

      // Central cavern rock arch & shafts
      { x: 800, y: 800, w: 600, h: 40, type: 'rock', color: '#5d4037' },
      { x: 650, y: 550, w: 900, h: 35, type: 'rock', color: '#5d4037' },
      { x: 900, y: 320, w: 400, h: 30, type: 'rock', color: '#6d4c41' },

      // Left cavern chambers
      { x: 150, y: 1100, w: 450, h: 30, type: 'rock', color: '#5d4037' },
      { x: 200, y: 820, w: 350, h: 30, type: 'rock', color: '#5d4037' },
      { x: 100, y: 500, w: 400, h: 30, type: 'rock', color: '#5d4037' },
      { x: 250, y: 250, w: 300, h: 25, type: 'one-way', color: '#8d6e63' },

      // Right cavern chambers
      { x: 1600, y: 1100, w: 450, h: 30, type: 'rock', color: '#5d4037' },
      { x: 1650, y: 820, w: 350, h: 30, type: 'rock', color: '#5d4037' },
      { x: 1700, y: 500, w: 400, h: 30, type: 'rock', color: '#5d4037' },
      { x: 1650, y: 250, w: 300, h: 25, type: 'one-way', color: '#8d6e63' },

      // Vertical shaft dividers
      { x: 600, y: 830, w: 35, h: 300, type: 'rock', color: '#4a3525' },
      { x: 1565, y: 830, w: 35, h: 300, type: 'rock', color: '#4a3525' }
    ],
    bushes: [
      { x: 1020, y: 280, w: 160, h: 40 },
      { x: 1020, y: 510, w: 160, h: 40 },
      { x: 320, y: 780, w: 140, h: 40 },
      { x: 1720, y: 780, w: 140, h: 40 },
      { x: 980, y: 1300, w: 240, h: 40 }
    ],
    spawns: [
      { x: 300, y: 440 },
      { x: 1900, y: 440 },
      { x: 1100, y: 260 },
      { x: 1100, y: 740 },
      { x: 350, y: 1040 },
      { x: 1850, y: 1040 }
    ],
    weaponSpawns: [
      { x: 1100, y: 280, weaponType: 'sniper' },
      { x: 1100, y: 510, weaponType: 'ar' },
      { x: 300, y: 460, weaponType: 'smg' },
      { x: 1900, y: 460, weaponType: 'sniper' },
      { x: 350, y: 1060, weaponType: 'ar' },
      { x: 1850, y: 1060, weaponType: 'smg' },
      { x: 1100, y: 760, weaponType: 'ar' }
    ],
    healthSpawns: [
      { x: 1100, y: 480 },
      { x: 250, y: 780 },
      { x: 1950, y: 780 }
    ]
  },
  hightower: {
    id: 'hightower',
    name: 'High Tower (Sky Battle)',
    width: 2400,
    height: 1500,
    backgroundColor: '#0c1b2b',
    skyColor: '#1a365d',
    platforms: [
      // Bottom Void Danger & Base Platform
      { x: 0, y: 1440, w: 2400, h: 60, type: 'rock', color: '#1a202c' },
      { x: 0, y: 0, w: 40, h: 1500, type: 'rock', color: '#1a202c' },
      { x: 2360, y: 0, w: 40, h: 1500, type: 'rock', color: '#1a202c' },

      // Center High Tower Structure
      { x: 1050, y: 200, w: 300, h: 1200, type: 'metal', color: '#2d3748' },
      { x: 950, y: 200, w: 500, h: 30, type: 'metal', color: '#4a5568' },
      { x: 900, y: 450, w: 600, h: 30, type: 'metal', color: '#4a5568' },
      { x: 850, y: 700, w: 700, h: 30, type: 'metal', color: '#4a5568' },
      { x: 800, y: 950, w: 800, h: 30, type: 'metal', color: '#4a5568' },
      { x: 750, y: 1200, w: 900, h: 35, type: 'metal', color: '#4a5568' },

      // Side Floating Sky Platforms (Left)
      { x: 250, y: 350, w: 350, h: 25, type: 'one-way', color: '#3182ce' },
      { x: 150, y: 650, w: 400, h: 25, type: 'one-way', color: '#3182ce' },
      { x: 300, y: 950, w: 350, h: 25, type: 'one-way', color: '#3182ce' },

      // Side Floating Sky Platforms (Right)
      { x: 1800, y: 350, w: 350, h: 25, type: 'one-way', color: '#3182ce' },
      { x: 1850, y: 650, w: 400, h: 25, type: 'one-way', color: '#3182ce' },
      { x: 1750, y: 950, w: 350, h: 25, type: 'one-way', color: '#3182ce' }
    ],
    bushes: [
      { x: 1120, y: 160, w: 160, h: 40 },
      { x: 350, y: 310, w: 140, h: 40 },
      { x: 1910, y: 310, w: 140, h: 40 },
      { x: 1120, y: 660, w: 160, h: 40 },
      { x: 1120, y: 1160, w: 160, h: 40 }
    ],
    spawns: [
      { x: 1200, y: 140 },
      { x: 400, y: 300 },
      { x: 2000, y: 300 },
      { x: 1200, y: 640 },
      { x: 300, y: 600 },
      { x: 2100, y: 600 },
      { x: 1200, y: 1140 }
    ],
    weaponSpawns: [
      { x: 1200, y: 160, weaponType: 'sniper' },
      { x: 1200, y: 410, weaponType: 'ar' },
      { x: 1200, y: 660, weaponType: 'smg' },
      { x: 400, y: 320, weaponType: 'ar' },
      { x: 2000, y: 320, weaponType: 'smg' },
      { x: 300, y: 620, weaponType: 'ar' },
      { x: 2100, y: 620, weaponType: 'sniper' },
      { x: 1200, y: 1160, weaponType: 'smg' }
    ],
    healthSpawns: [
      { x: 1200, y: 900 },
      { x: 350, y: 910 },
      { x: 1950, y: 910 }
    ]
  }
};
